libsql/internal/http/hranaV2/hranaV2.go

```diff
@@ -128,67 +132,83 @@ func (h *hranaV2Conn) sendPipelineRequest(ctx context.Context, msg *hrana.Pipeli
 		// If the stream is closed, we can't send any more requests using this connection.
 		return nil, fmt.Errorf("stream is closed: %w", driver.ErrBadConn)
 	}
+	if h.closeStreamBeforeNextRequest {
+		msg.AddToFront(hrana.CloseStream())
+	}
 	if h.baton != "" {
 		msg.Baton = h.baton
 	}
-	reqBody, err := json.Marshal(msg)
+	result, streamClosed, err := sendPipelineRequest(ctx, msg, h.url, h.jwt, h.host)
+	if streamClosed {
+		h.streamClosed = true
+	}
 	if err != nil {
 		return nil, err
 	}
+	h.baton = result.Baton
+	if h.closeStreamBeforeNextRequest {
+		result.Results = result.Results[1:]
```

Comment:
Can the response to close and execute arrive out of order from the server?

Reply:
No. This is really a single request that is handled synchronously and response is sent back for the whole batch at once.